### PR TITLE
Replace Class::Load with Module::Runtime

### PR DESCRIPTION
**Pull request by xsawyerx**
This addresses #1145.

Module::Runtime is a faster module than Class::Load (even with
Class::Load::XS available) and is already available by our stack.

If and when App::Cmd removes its dependency on Class::Load, we
will not require this by our stack at all.

### CODE REVIEW

**Comment by racke** [comment]
:+1:

File: lib/Dancer2.pm [hunk 1]
```diff
@@ -85,7 +85,7 @@ sub import {
     $final_args{dsl} ||= $config_dsl;
 
     # load the DSL, defaulting to Dancer2::Core::DSL
-    load_class( $final_args{dsl} );
+    use_module( $final_args{'dsl'} );
     my $dsl = $final_args{dsl}->new( app => $app );
     $dsl->export_symbols_to( $caller, \%final_args );
```
Comment (veryrusty) [inline lib/Dancer2.pm]: This could be chained:

`my $dsl = use_module(  $final_args{'dsl'} )->new( app => $app );`

Comment (xsawyerx) [inline lib/Dancer2.pm]: D'oh! I intended to do that! @veryrusty++

Comment (xsawyerx) [inline lib/Dancer2.pm]: Done!

**Comment by SysPete** [comment]
:+1:

**Comment by veryrusty** [comment]
We _try_ to maintain compatibility back to perl 5.8.n. Module::Runtime works around a couple of core bugs before the 5.11 release. 

Should we be using `require_module` instead of `eval { require Some::Module; 1 }` to take advantage of that ?

**Comment by xsawyerx** [comment]
@veryrusty added, but only for `lib`, not `t`.

**Comment by veryrusty** [comment]
@xsawyerx++   :+1: with the `require_module` changes.

**Comment by veryrusty** [comment]
Merged as 1e97ed3a. Thanks all!

**Comment by xsawyerx** [comment]
Thank you. :)

**Pull request closed without merging**